internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -970,30 +1000,54 @@ func modifyMappedVirtualDisk(
 	rt guestrequest.RequestType,
 	mvd *guestresource.LCOWMappedVirtualDisk,
 	securityPolicy securitypolicy.SecurityPolicyEnforcer,
+	isContainerBlockDevice bool,
 ) (err error) {
 	var verityInfo *guestresource.DeviceVerityInfo
-	if mvd.ReadOnly {
-		// The only time the policy is empty, and we want it to be empty
-		// is when no policy is provided, and we default to open door
-		// policy. In any other case, e.g. explicit open door or any
-		// other rego policy we would like to mount layers with verity.
-		if len(securityPolicy.EncodedSecurityPolicy()) > 0 {
-			devPath, err := scsi.GetDevicePath(ctx, mvd.Controller, mvd.Lun, mvd.Partition)
-			if err != nil {
-				return err
-			}
-			verityInfo, err = verity.ReadVeritySuperBlock(ctx, devPath)
-			if err != nil {
-				return err
-			}
+
+	// enforcePolicy specifies if the device mount and unmount should be
+	// enforced via policy framework.
+	//
+	// We enforce the mount_device and unmount_device enforcement
+	// point only for read-only devices.
+	// For read-write devices, we cannot use the verity hash for
+	// policy enforcement and therefore, we skip any enforcement
+	// for the same. That being said, we perform a check outside
+	// the policy framework to ensure that the device is being
+	// mounted in an expected location.
+	enforcePolicy := mvd.ReadOnly
```

Comment:
We should always enforce a policy for SCSI devices. The fact that we currently don't for RW SCSI is not ideal and it's in our backlog to address.
We also shouldn't rely on an external parameter either, i.e. `isContainerBlockDevice` and however it's set outside of this function.
We can (e.g.) change the logic to always read verity block first, when security policy is present:
- If verity block exists - then this must be a container layer, we'll enforce device hash policy.
- If verity block is absent - then this must be a block device (the only supported non-layer SCSI attachment type), and we enforce that policy. As part of this enforcement we can check for RO/RW flags, also, we add a device cgroup for this attachment, which should also be considered by policy enforcement.